internal/openshift/client/openshift_client.go

```diff
@@ -143,6 +145,56 @@ func (o openShift) Idle(apiURL string, bearerToken string, namespace string, ser
 	return
 }
 
+// Reset deletes a pod and start a new one
+func (o openShift) Reset(apiURL string, bearerToken string, namespace string) (err error) {
+	logger.Info("reseting pods in " + namespace)
+
+	req, err := o.reqAPI(apiURL, bearerToken, "GET", namespace, "pods", nil)
+	if err != nil {
+		return err
+	}
+	resp, err := o.do(req)
+	if err != nil {
+		return err
+	}
+
+	defer bodyClose(resp)
+
+	podList := &v1.PodList{}
+	err = json.NewDecoder(resp.Body).Decode(podList)
+	if err != nil {
+		return err
+	}
+
+	for _, element := range podList.Items {
+
+		podName := element.GetName()
+		if strings.Contains(podName, "deploy") {
+			continue
+		}
+
+		log.Infof("Reseting the pod " + podName)
+		req, err := o.reqAPI(apiURL, bearerToken, "DELETE", namespace, "pods/"+podName, nil)
```

Comment:
may want to take inspiration from their api and  use `DELETE` verb for our api endpoint as well.

Reply:
`DELETE` is misinterpreting. We are not interested in getting rid of the pod. We are interested in deleting this one and starting a new one. I think `reset` is more suitable here, but I don't care much about the name so if you feel I should change it to `delete`, I will do so.

Reply:
no .. you are right. don't use delete. I think it must be POST atleast so that we know it is unsafe.